pkg/cli/cli.go

```diff
@@ -131,6 +135,7 @@ func newCLI(options ...Option) (*CLI, error) {
 		defaultProjectVersion: cfgv3.Version,
 		defaultPlugins:        make(map[config.Version][]string),
 		plugins:               make(map[string]plugin.Plugin),
+		fs:                    afero.NewOsFs(),
```

Comment:
Right now there's no way for a user to set `fs`, so perhaps an Option to do so?

Reply:
Do we really want to expose that? I think that we can internally use that field to set a `MemMapFs` or even achieve atomicity with a `CopyOnReadFs`, but what advantage would a CLI tool developer get from exposing that field?

Either way, creating an `Option` would be incremental, and therefore non-breaking, so I think we can leave that for a follow up after we discuss it.

Reply:
True that wouldn't be breaking. Sounds good.